scipy/special/cephes/stirling.c

```diff
@@ -0,0 +1,73 @@
+/* c implementation of Stirling numbers of the second kind */
+
+/*
+ *
+ *     Stirling numbers of the second kind
+ *
+ *
+ *
+ * SYNOPSIS: Stirling numbers of the second kind count the
+ *  number of ways to make a partition of n distinct elements
+ *  into k non-empty subsets.
+ *
+ * DESCRIPTION: n is the number of distinct elements of the set
+ *  to be partitioned and k is the number of non-empty subsets.
+ *  The values for n < 0 or k < 0 are interpreted as 0. If you
+ *
+ * ACCURACY: The method returns an unsigned integer type and
+ *  the size of the type 32 bits.
+ *
+ *
+ * NOTE: this file is *NOT* part of the ceph distribution and was
+ *  added by Lucas Roberts in 2023 after opening this issue
+ *  https://github.com/scipy/scipy/issues/17890
+ */
+#include "mconf.h"
+#include <stdlib.h>
```

Comment:
```suggestion
#include "mconf.h"
#include <limits.h>
#include <stdlib.h>
```

We'll need this to check for overflow.